### PR TITLE
Adding USM support for Level 2 operators in SYCL-BLAS

### PR DESCRIPTION
**Pull request by muhammad-tanvir-1211**
This PR extends the USM support in SYCL-BLAS for Level 2 operators. All Level 2 benchmarks and tests now use USM and Buffers.

FYI: This PR is blocked by #385

### CODE REVIEW

**Review by aacostadiaz — COMMENTED** [review]
LGTM. I have exactly the same comments that I had in the Level 1 PR. If you make any changes in the level 1 PR to address these comments please propagate the changes to this PR as well.

**Review by aacostadiaz — APPROVED** [review]
LGTM.

The PR is quite large but it is just updating the level 2 operators to work with the changes introduced in the previous pull request.

File: src/interface/blas2/spr.cpp.in [hunk 1]
```diff
@@ -36,7 +36,7 @@ template typename SB_Handle::event_t
 _spr<SB_Handle, ${INDEX_TYPE}, ${DATA_TYPE}, ${container_t0}, ${INCREMENT_TYPE},
      ${container_t1}>(SB_Handle& sb_handle, char _Uplo, ${INDEX_TYPE} _N,
                       ${DATA_TYPE} _alpha, ${container_t0} _vx,
-                      ${INCREMENT_TYPE} _incx, ${container_t1} _mPA);
+                      ${INCREMENT_TYPE} _incx, ${container_t1} _mPA, const typename SB_Handle::event_t& _dependencies);
```
Comment (pgorlani) [inline src/interface/blas2/spr.cpp.in:39]: Thanks for your patch, @muhammad-tanvir-1211.

Just a minor thing, the code format didn't consider the `.cpp.in` files.

```suggestion
                      ${INCREMENT_TYPE} _incx, ${container_t1} _mPA,
                      const typename SB_Handle::event_t& _dependencies);
```

**Review by pgorlani — APPROVED** [review]

File: src/interface/blas2/spr.cpp.in (hunk 1 above)
Comment (s-Nick) [reply src/interface/blas2/spr.cpp.in:39]: quick note on this file. `spr` is the only interface that has template parameter explicitly written. I think we can remove them and it should work fine.

**Pull request merged**